draft-schinazi-quic-version-negotiation.md

```diff
@@ -28,78 +28,81 @@ author:
+<http://github.com/ekr/draft-schinazi-quic-version-negotiation>.
 
 
 # Conventions and Definitions
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this
-document are to be interpreted as described in BCP 14 {{RFC2119}} {{RFC8174}}
+document are to be interpreted as described in BCP 14 {{!RFC2119}} {{!RFC8174}}
 when, and only when, they appear in all capitals, as shown here.
 
 
 # Version Negotiation Mechanism
 
 The mechanism defined in this document is straightforward: the client maintains
 a list of QUIC versions it supports, ordered by preference. Its Initial packet
-is sent using the version that the server is most likely to support (in
-practice, this will generally be the oldest version the client supports);
-that Initial packet then lists all of the other compatible versions
-({{compatible-versions}}) that the client supports in the
-supported_compatible_versions field of its transport parameters
-({{compat-vers-tp}}). The server then selects its preferred version and
-responds with that version in all of its future packets (except for Retry, as
-below). It also inserts the selected version in the
-negotiated_compatible_version field of its transport parameters.
-
-The server MUST NOT select a version not offered by the client.  The client MUST
-validate that the version in the server's packets is one of the versions that it
-offered and that it matches the value in the server's transport parameters.
+is sent using the version that the server is most likely to support (in the
+absence of other information, this will often be the oldest version the client
+supports); that Initial packet then lists all of the other compatible versions
+({{compatible-versions}}) that the client supports in the Compatible Version
+fields of its transport parameters ({{fig-client-tp}}).
+
+1. If the server supports the client's attempted version, and supports one of
+   the client's compatible versions, it selects the first version it supports
+   from the client's compatible version list. It then responds responds with
+   that version in all of its future packets (except for Retry, as below).
+
+1. If the server supports the client's attempted version but does not support
+   any of the client's compatible versions, it proceeds with the clients
+   attempted version.
+
+1. If the server does not support the client's attempted version, it sends a
+   Version Negotiation packet listing all the versions it supports.
+
+If the server leverages compatible versions and responds with a different
+version from the client's attempted version, it MUST NOT select a version not
+offered by the client.  The client MUST validate that the version in the
+server's packets is either its attempted version, or one of the compatible
+versions that it offered and that it matches the negotaited version in the
```

Comment:
misspell of negotiated. Also, see above about having the client's list being exhaustive.

Reply:
Fixed